### ISSUE TITLE
some questions of Singularity

### ISSUE THREAD

**Issue by sheldon-pan** [apptainer/singularity#4843]
### Version of Singularity:

What version of Singularity are you using? Run:

```
$ singularity version
3.4.2-1.el7
```

### Expected behavior

What did you expect to see when you do...?
1 If I want to use pytorch in singularity,should I install CUDA and cudnn in img?
2 Can I build an offline repo in the HPC  environment.
3 Does SIF file have a distribution mechanism in clusters?
4 Why does the singularity use the layer method like docker? So I must download full sif image if I want to use  python or tensorflow? 

### Actual behavior

What actually happend? Why was it incorrect?

### Steps to reproduce this behavior

How can others reproduce this issue/problem?

### What OS/distro are you running

```
$ cat /etc/os-release

centos 7.4 1708
```

### How did you install Singularity

Write here how you installed Singularity. Eg. RPM, source.
source

**Comment by dtrudg** [comment]
> 1 If I want to use pytorch in singularity,should I install CUDA and cudnn in img?

Take a look at the guide here which explains how things work a bit: https://sylabs.io/guides/3.5/user-guide/gpu.html

The basic CUDA drivers/libraries should be installed on the host. CuDNN should go into the container.

> 2 Can I build an offline repo in the HPC environment.

Can you give more information on what you want to accomplish here? I'm not sure what you mean by offline repo.

> 3 Does SIF file have a distribution mechanism in clusters?

Since SIF is a simple single file, it will generally be made available on a cluster by placing it on a shared filesystem. The Sylabs Container library (https://cloud.sylabs.io) is available for broader sharing.

> 4 Why does the singularity use the layer method like docker? So I must download full sif image if I want to use python or tensorflow?

Using layers has the advantage of saving bandwidth and space when storing things used by many containers. However, using the single file SIF format has benefits including:

- easy sharing - it's just a single file to copy
- an immutable run-time format
- cryptographic signing that is verifiable at runtime
- encryption via standard kernel interfaces
